_posts/2022-04-10-using-log-collection.markdown

```diff
@@ -0,0 +1,156 @@
+---
+layout: post
+title:  "Log Collection on Flotta devices"
+date:   2022-04-10 12:00:00 +0100
+categories: flotta
+author: Eloy Coto <eloycoto@acalustra.com>
+tags:
+  - guide
+  - flotta
+  - logs
+
+summary-1: Introduction to flotta log collection
+---
+
+When running thousands of devices on the far edge, log collection retrieval
+maybe differs from one site to another, as also the limit of data that can be
+sent from the device or stored in the device.
+
+At the same time, workloads logs can vary in terms of importance, so the backup
+workload logs maybe need to be always pushed, and maybe another workload with
+less importance can lose logs on network connectivity issues. At the same time,
+maybe you don’t want that workload that manages any personal information sends
+the information outside the country.
+
+For that, Flotta introduces log collection targets, so each device - or
+deviceSets- will have a set of custom logs servers that one workload can use to
+send the logs.
+
+![Device Logs diagram](./assets/images/DeviceLogsConfig.png)
+
+So the main use case is to be able to define servers in edge devices and be
+consumed by the workloads, so in this way, the roles will be limited between
+developers and edge administrators.
+
+*A sample workflow for a retail store can be:*
+
+**Edge Device Administrator:** defines the device security, heartbeat and
+general options based on the device or deviceSet. In this case, will define two
+kinds of logs servers, LogSecure which always push the logs to the server, with
+big storing buffers. Meanwhile, there is a logGeneral, that it’s the server
+that may be lost logs on networks downtimes, or maybe using different
+transports.
+
+**Edge developer:** In this case, the developer knows that for running the
+workload in all stores, there are always two kinds of log servers that can be
+used, because Edge Administration shared all the details about buffer, quotas
+and destination.
+
+To collect logs, a subprocess reads the workloads stdout and pushed it to a
+buffer, even if the device is network disconnected for a while, it’ll push the
+right timestamp to the server, due to the buffer knows when the log was
+retrieved. When the buffer is full, some logs will be dropped to avoid filling
+the RAM on the device.
+
+Regarding transports, at the moment, the Flotta device agent only supports raw
+Syslog transport, but in the following releases, new transports will be added,
+like Loki, Kafka, Elastic and MQTT.
+
+## Example setup
+
+In terms of edge device setup, it’s quite easy, for the case of Syslog, the
```

Comment:
"It's quite easy to setup an edge device; in case of Syslog, the following config-maps should be in place"